force-app/main/default/classes/PermissionSetMapper.cls

```diff
@@ -0,0 +1,75 @@
+      from this software without specific prior written permission.
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2021
+* @group Permission
+* @group-content ../../ApexDocContent/Permission.htm
+* @description Mapper class for Permission Set.
+*/
+public virtual with sharing class PermissionSetMapper {
+    
+    /*********************************************
+    * @description Instance for Singleton Pattern
+    **********************************************/
+    @TestVisible
+    private static PermissionSetMapper instance;
+
+    /*********************************************************************
+    * @description Empty constructor is not public for Singleton Pattern
+    **********************************************************************/
+    protected PermissionSetMapper() {}
+
+    /*****************************************************************************
+    * @description Static method to get the current instance for Singleton pattern
+    * @return The instance of PermissionSetMapper.
+    ******************************************************************************/
+    public static PermissionSetMapper getInstance() {
+        if (instance == null) {
+            instance = new PermissionSetMapper();
+        }
+
+        return instance;
+    }
+    
+    /*****************************************************************************
+    * @description Check if provided Permission set has access to view Setup.
+    * @param permissionSetIds A set of Permission set ids.
+    * @return If provided permission set has access to view Setup.
+    ******************************************************************************/
+    public virtual Boolean checkPermissionSetsWithViewSetupFromIds(Set<Id> permissionSetIdsSet) {
+        List<PermissionSet> permissionSets = [SELECT Id 
+                                            FROM PermissionSet
+                                            WHERE Id in :permissionSetIdsSet
+                                                AND PermissionsViewSetup = true];
+        if (permissionSets.size() > 0) {
+            return true;
+        }
+
+        return false;
```

Comment:
Why don't we return `!permissionSets.isEmpty()` or `permissionsSets.isEmpty() == false`?